File: databricks/notebooks/tools/Find_Best_Run.py
# Databricks notebook source
# MAGIC %md ### Find Best Run of an experiment
# MAGIC 
# MAGIC * Finds the best run of an experiment by searching for the min or max of a metric.
# MAGIC * Uses: [Best_Run_Utils](https://demo.cloud.databricks.com/#notebook/2960189)
# MAGIC 
# MAGIC Samples:
# MAGIC * 2100777 - auroch - min
# MAGIC * 2386160 - max_bins

# COMMAND ----------

# MAGIC %md #### Setup

# COMMAND ----------

# MAGIC %run ../common/Best_Run_Utils

# COMMAND ----------

default_experiment_id = "2630330" # /Shared/experiments/demo/sk_WineQuality_demo
default_metric = "rmse"

# COMMAND ----------

dbutils.widgets.text("Experiment ID", default_experiment_id)
dbutils.widgets.text("Metric", default_metric)
dbutils.widgets.dropdown("isLargerBetter","no",["yes","no"])

experiment_id = int(dbutils.widgets.get("Experiment ID"))
metric = dbutils.widgets.get("Metric")
isLargerBetter = dbutils.widgets.get("isLargerBetter") == "yes"

experiment_id,metric,isLargerBetter

# COMMAND ----------

# MAGIC %md #### Calls to find best run

# COMMAND ----------

get_best_run(experiment_id,metric,isLargerBetter)

# COMMAND ----------

# MAGIC %md #### Call to find last run

# COMMAND ----------

get_last_run(experiment_id)

File: databricks/notebooks/tools/scala_tools/Find_Best_Run.scala
// Databricks notebook source
// MAGIC %md ### Find Best Run of an experiment
// MAGIC 
// MAGIC * Finds the best run of an experiment by searching for the min or max of a metric.
// MAGIC * Note: Scala version is not up to par with Python version.

// COMMAND ----------

// MAGIC %run ./BestRunUtils

// COMMAND ----------

val defaultExperiment_id = "2630330" // /Shared/experiments/demo/sk_WineQuality_demo
val defaultMetric = "rmse"

// COMMAND ----------

dbutils.widgets.text("Experiment ID", defaultExperiment_id)
dbutils.widgets.text("Metric", defaultMetric)
dbutils.widgets.dropdown("isLargerBetter","no",Seq("yes","no"))

val experimentId = dbutils.widgets.get("Experiment ID")
val metric = dbutils.widgets.get("Metric")
val isLargerBetter = dbutils.widgets.get("isLargerBetter") == "yes"

// COMMAND ----------

import org.mlflow.tracking.MlflowClient
val client = new MlflowClient()

// COMMAND ----------

val best = BestRunUtils.getBestRun(client, experimentId,metric,isLargerBetter)

// COMMAND ----------

// MAGIC %run ./DumpRun.scala

// COMMAND ----------

val runId = best.run.getInfo.getRunId

// COMMAND ----------

DumpRun.dumpRun(client, runId, 0)